### ISSUE TITLE
got the proxy error when using selenium-wire

### ISSUE THREAD

**Issue by beibeiqi0916** [wkeeling/selenium-wire#85]
I am running the test suits by switching between selenium-wire and selenium, but looks like after the test cases triggered by selenium-wire completed, then the next url triggered by selenium is broken, say ERR_PROXY_CONNECTION_FAILED. 
I can see the last step in log for selenium-wire is 'INFO:seleniumwire.proxy.client:Destroying proxy', just guess that might be the reason.
So are there any solutions I can check with?

**Comment by wkeeling** [comment]
Thanks for raising this. The driver instances should be independent of one another. Are you able to post the code that instantiates the selenium and selenium-wire webdrivers from your test?

**Comment by beibeiqi0916** [comment]
Sure, I pass the wire as the parameter, so
  `if wire:
         driver = CustomWireChrome(executable_path, chrome_options)
else:
         driver = CustomChrome`
and I define CustomWireChrome is separate class
`from seleniumwire import webdriver as wiredriver
class CustomWireChrome(WebDriverExtensions, wiredriver.Chrome):
    pass`

**Comment by YourMan** [comment]
I got the same error on Chrome. Did you find a solution, by any chance?

**Comment by fantik11** [comment]
Same error in both Chrome and Firefox. Any solution?

**Comment by wkeeling** [comment]
I think the original issue is due to the two driver instances sharing the same `desired_capabilities` object (a dictionary).  When the Selenium Wire instance `CustomWireChrome` is created, it modifies the desired capabilities by adding its internal proxy configuration to it. That means that the regular Selenium instance also receives this proxy configuration, but by the time that driver executes, the Selenium Wire proxy has disappeared and you see the error.

I'll look at fixing this in Selenium Wire so that the desired capabilities don't get "polluted" with this internal proxy config. In the meantime, it's probably a good idea to ensure that the two driver instances receive separate copies of the desired capabilities, rather than sharing the same instance. For example:

```
import copy

if wire:
    driver = CustomWireChrome(executable_path=settings.CHROMEDRIVER_PATH,
                          chrome_options=chrome_profile,
                          desired_capabilities=copy.deepcopy(dc))  # Use a copy
else:
    driver = CustomChrome(executable_path=settings.CHROMEDRIVER_PATH,
                          chrome_options=chrome_profile,
                          desired_capabilities=dc)
```

**Comment by wkeeling** [comment]
A copy of the desired capabilities is now made before it's used which eliminates side-effects caused by sharing. Available in v3.0.0 onwards.
